vm/actor/src/builtin/market/mod.rs

```diff
@@ -381,29 +380,36 @@ impl Actor {
     {
         rt.validate_immediate_caller_type(std::iter::once(&*MINER_ACTOR_CODE_ID))?;
         let miner_addr = *rt.message().caller();
+        let curr_epoch = rt.curr_epoch();
 
         let st: State = rt.state()?;
+        let proposals = DealArray::load(&st.proposals, rt.store()).map_err(|e| {
+            e.downcast_default(ExitCode::ErrIllegalState, "failed to load deal proposals")
+        })?;
 
-        let (deal_weight, verified_deal_weight, deal_space) = validate_deals_for_activation(
-            &st,
-            rt.store(),
-            &params.deal_ids,
-            &miner_addr,
-            params.sector_expiry,
-            params.sector_start,
-        )
-        .map_err(|e| {
-            e.downcast_default(
-                ExitCode::ErrIllegalState,
-                "failed to validate deal proposals for activation",
+        let mut weights = Vec::with_capacity(params.sectors.len());
```

Comment:
this will panic, although the capacity is set here, the length needs to be set as you are indexing below. Either you need to initialize the vector with that length before iterating, or instead of indexing below just push to the vector below. I recommend the latter to avoid indexing, but up to you.

If the go initialization is confusing, it's
```
make(<type>, <length>, <capacity>)
```

so they are specifically sizing the slice in go, which is why they can index it.

Reply:
ah, yeah that is a whoopsies